src/containers/Application/ApplicationCreate.tsx

```diff
@@ -47,33 +52,53 @@ const ApplicationCreate: React.FC = () => {
       sections={templateSections}
       handleClick={() => {
         setApplicationState({ type: 'reset' })
-        handleCreateApplication(applicationMutation, setApplicationState, templateType)
+        handleCreate({
+          create,
+          setApplicationState,
+          templateName: templateType.name,
+          templateId: templateType.id,
+          templateSections: templateSections.map((section) => {
+            return { templateSectionId: section.id }
+          }),
+          templateResponses: templateElementsIds.map((id) => {
+            return { templateElementId: id }
+          }),
+        })
       }}
     />
   ) : (
     <Header as="h2" icon="exclamation circle" content="No template found!" />
   )
 }
 
-function handleCreateApplication(
-  applicationMutation: any,
-  setApplicationState: any,
-  template: TemplateTypePayload
-) {
+interface handleCreateProps {
+  create: (props: createApplicationProps) => void
+  setApplicationState: any
+  templateName: string
+  templateId: number
+  templateSections: { templateSectionId: number }[]
+  templateResponses: { templateElementId: number }[]
+}
+
+function handleCreate({
```

Comment:
At this point, I am not sure what is the benefit f moving handleCreate method outside of AppicationCreate component.

If it's a scope method inside ApplicationCreate, I think it would be easier to digest the code (and less lines).

Also it would expose things like this logic:

```
setApplicationState({ type: 'reset'})
setApplicationState({ type: 'setSerialNumber', serialNumber })
```

Reply:
Yes... I was trying to follow your original idea to keep everything out of the main component. So it would only have the return block with some variables and hook defined at the top. Bu I agree that it would be easier to read if this were inside. I'll move back in the component.